libfwupd/fwupd-enums.h

```diff
@@ -94,6 +94,9 @@ typedef enum {
  * @FWUPD_DEVICE_FLAG_WILL_DISAPPEAR:		Device will disappear after update and can't be verified
  * @FWUPD_DEVICE_FLAG_CAN_VERIFY:		Device checksums can be compared against metadata
  * @FWUPD_DEVICE_FLAG_CAN_VERIFY_IMAGE:		Image can be dumped from device for verification
+ * @FWUPD_DEVICE_FLAG_CAN_SELF_RECOVER:		Device update architecture has resiliency abilities:
+ * 						 * Multiple software banks changed only after success
+ * 						 * Will accept correct firmware image in bootloader
```

Comment:
in all honesty, I think you're trying to encode two bits of information as one thing. Perhaps you want two things, `FLAG_IS_DUAL_IMAGE` or something for the A/B partition (which can be used by coreboot too) and `FLAG_BOOTLOADER_UNIQUE`or `FLAG_BOOTLOADER_RECOVER` to say that the bootloader is clever enough to know what firmware it needs rather than being some generic device like most DFU devices are. For the second part, would the bootloader need to know the version of the runtime, or is a unique VID/PID enough? For instance, unifying receivers know what kind of firmware they need, but the version is faked to `0` so that any firmware version can be used to recover them.

Reply:
Yes they're two bits of information; but I was thinking specifically about how it would be used.
The daemon won't care about this information.  It should chug along either way.
My intent is that when someone has configured policy as such a client could install firmware in the background on safe devices.

Realistically clients shouldn't do firmware upgrade in the background unless a device has both aspects.

For the second I think of it as the right GUID is populated even in bootloader mode so the right firmware automatically installs without a user needing to do something.
This can mean a few things:
* Device has the correct VID/PID in bootloader mode (not a bootloader VID/PID)
* Device populated VID/PID from a hardware platform hint (ACPI, GPIO, etc)

Reply:
I don't think fwupd should be the one that defines policy about whether it's "safe" -- I think fwupd should give the client the information about the device and then we can define safe there. _Safe_ might change depending on AC status or if the device is removable and thus end-user replaceable.

Reply:
OK.  I've split into two flags.